### PR TITLE
 [mort] Bring back mort

### PR DESCRIPTION
**Pull request by ebraminio**
Started by reverting https://github.com/harfbuzz/harfbuzz/commit/1f1c85a5

Just a starting point, if we agree even mort can come back.

No breakage on `morx` and `kerx` tests at least.

### CODE REVIEW

File: src/hb-aat-layout.cc [hunk 1]
```diff
@@ -130,9 +130,23 @@ hb_aat_layout_find_feature_mapping (hb_tag_t tag)
 
 
 /*
- * morx/kerx/trak
+ * mort/morx/kerx/trak
  */
 
+// static inline const AAT::mort&
```
Comment (ebraminio) [inline src/hb-aat-layout.cc]: Issues arise on enabling this :)

File: src/hb-aat-layout-morx-table.hh [hunk 2]
```diff
@@ -69,12 +73,12 @@ struct RearrangementSubtable
 	ret (false),
 	start (0), end (0) {}
 
-    inline bool is_actionable (StateTableDriver<EntryData> *driver HB_UNUSED,
+    inline bool is_actionable (StateTableDriver<MorxTypes, EntryData> *driver HB_UNUSED,
```
Comment (behdad) [inline src/hb-aat-layout-morx-table.hh]: Why do these say MorxTypes instead of Types?

File: src/hb-aat-layout-morx-table.hh [hunk 3]
```diff
@@ -830,18 +845,21 @@ struct ChainSubtable
...
-  InsertionSubtable		insertion;
+  RearrangementSubtable<Types>	rearrangement;
+  ContextualSubtable<Types>	contextual;
+  LigatureSubtable<Types>	ligature;
+  NoncontextualSubtable<Types>	noncontextual;
+  InsertionSubtable<Types>	insertion;
   } u;
   public:
-  DEFINE_SIZE_MIN (12);
+  DEFINE_SIZE_MIN (2 * sizeof (HBUINT32) + 4);
```
Comment (behdad) [inline src/hb-aat-layout-morx-table.hh]: What's with this line?

File: src/hb-aat-layout-morx-table.hh [hunk 4]
```diff
@@ -950,38 +968,39 @@ struct Chain
...
+  HBUINT	featureCount;	/* Number of feature subtable entries. */
+  HBUINT	subtableCount;	/* The number of subtables in the chain. */
 
   UnsizedArrayOf<Feature>	featureZ;	/* Features. */
 /*ChainSubtable	firstSubtable;*//* Subtables. */
 /*subtableGlyphCoverageArray*/	/* Only if version >= 3. We don't use. */
 
   public:
-  DEFINE_SIZE_MIN (16);
+  DEFINE_SIZE_MIN (2 * sizeof (HBUINT) + 4);
```
Comment (behdad) [inline src/hb-aat-layout-morx-table.hh]: And this?  Specially since it's changing the number.

Comment (ebraminio) [reply src/hb-aat-layout-morx-table.hh]: Fixed, more work is needed however

File: src/hb-aat-layout-morx-table.hh (hunk 3 above)
Comment (ebraminio) [reply src/hb-aat-layout-morx-table.hh]: Fixed, more work is needed however

File: src/hb-aat-layout-morx-table.hh (hunk 2 above)
Comment (ebraminio) [reply src/hb-aat-layout-morx-table.hh]: Ops, replace error

File: src/hb-aat-layout.cc (hunk 1 above)
Comment (ebraminio) [reply src/hb-aat-layout.cc]: Fixed

**Comment by behdad** [comment]
I already cherry-picked yours and am ten commits in.

**Pull request closed without merging**